File: Data Structures & Algorithms/binary-search/submission-3.java
class Solution {
    public int search(int[] nums, int target) {
        int L = 0;
        int R = nums.length - 1;

        while (L <= R){
            int C = (L+R)/2;
            if(nums[C] == target){
                return C;
            }else if(target < nums[C]){
                R = C - 1;
            }else if(target > nums[C]){
                L = C + 1;
            }
        }
        return -1;
    }
}

File: Data Structures & Algorithms/search-2d-matrix/submission-3.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        # iterate through arrays, check curlist[-1]
        # if curlist[-1] < target, move right
        # if curlist[-1] > target inspect current array
        # if curlist[-1] == target, return 

        for curlist in matrix:
            if curlist[-1] < target:
                continue
            else:
                l, r = 0, len(curlist) - 1
                while l <= r:
                    mid = (l + r) // 2
                    if curlist[mid] > target:
                        r = mid - 1
                    elif curlist[mid] < target:
                        l = mid + 1
                    else:
                        return True
                return False
        return False
                    
